### PR TITLE
Make use of Jenkins Log Parser plugin

### PR DESCRIPTION
**Pull request by aspiers**
Take advantage of the Jenkins Log Parser plugin:

- https://wiki.jenkins-ci.org/display/JENKINS/Log+Parser+Plugin

This requires a patch to jenkins-job-builder in order for the per-project rules to be correctly activated during upload of the job to Jenkins:

- https://review.openstack.org/#/c/445509/

(However, it should not break anything if the patch is not applied.)

Once correctly configured, when viewing a job in the web UI, you can click on "Parsed Console Output" to see a nice analysis of the errors and warnings, broken down by mkcloud section.  Note that #1755 has already been merged to help clean up this UI.

Also remove the email publisher which breaks with the latest `jjb`.

### CODE REVIEW

File: jenkins/ci.suse.de/openstack-mkcloud.yaml [hunk 1]
```diff
@@ -283,10 +283,25 @@
...
         shopt -s extglob
+
+        echo $WORKSPACE
+        pwd
+
         # emptying the workspace
         mkdir -p empty
         rsync -r --delete empty/ ./
 
+        cat <<'EOF' > log-parser-plugin-rules.txt
```
Comment (jdsn) [inline jenkins/ci.suse.de/openstack-mkcloud.yaml]: Lets please maintain this as separate file in the git repo.
I am fine with having something like `cp $automationrepo/$something/log-parser-plugin-rules.txt .` here though/

File: jenkins/ci.suse.de/openstack-mkcloud.yaml [hunk 2]
```diff
@@ -283,10 +283,25 @@
...
+        start /^(===+> MKCLOUD STEP START: |MKCLOUD step: )/
```
Comment (jdsn) [inline jenkins/ci.suse.de/openstack-mkcloud.yaml]: I thought that you only wanted to match this: `MKCLOUD step:`
With the marker in as well there are too many sections now, or what am I missing?

File: jenkins/ci.suse.de/openstack-mkcloud.yaml [hunk 3]
```diff
@@ -283,10 +283,25 @@
     - shell: |
         set -x
         shopt -s extglob
+
+        echo $WORKSPACE
+        pwd
```
Comment (jdsn) [inline jenkins/ci.suse.de/openstack-mkcloud.yaml]: Do we need this debug output?

Comment (jdsn) [inline jenkins/ci.suse.de/openstack-mkcloud.yaml]: jenkins is showing it already before it starts the script

File: jenkins/ci.suse.de/openstack-mkcloud.yaml (hunk 2 above)
Comment (aspiers) [inline jenkins/ci.suse.de/openstack-mkcloud.yaml]: No there are not too many sections, it either matches one line of the new style, or one line of the old style, not both.  It's there for backwards compatibility.

File: jenkins/ci.suse.de/openstack-mkcloud.yaml [hunk 4]
```diff
@@ -283,6 +283,7 @@
     - shell: |
         set -x
         shopt -s extglob
+
```
Comment (bmwiedemann) [inline jenkins/ci.suse.de/openstack-mkcloud.yaml]: unrelated empty line
